pkg/router.go

```diff
@@ -159,11 +161,48 @@ func PointWrite(m *nmodule.Module, r *router.Request) ([]byte, error) {
 	}
 
 	// TODO: Encode PointWriter (pw)
+	serialData := endec.NewSerialData()
+	endec.SetPositionalData(serialData, true)
+	endec.SetRequestData(serialData, true)
+	msgId, _ := endec.GenerateRandomId()
+	endec.SetMessageId(serialData, msgId)
+	endec.UpdateBitPositionsForHeaderByte(serialData)
+
 	log.Infof("Point Address UUID >>>>>>> %s ", nstring.DerefString(pnt.AddressUUID))
 
-	// TODO: Encrypt the encoded data by using encrypter.Encrypt method
+	// If the PointWriter struct has fields, you can also print them individually
+	// Assuming pw has fields like `Name` and `Value`
+	for _, value := range *pw.Priority {
+		if value != nil {
+			floatValue := *value
+			if pnt.Name == "operation" {
+				endec.EncodeData(serialData, floatValue, endec.MDK_BOOL, 1)
+			} else if pnt.Name == "mode" {
+				endec.EncodeData(serialData, uint8(floatValue), endec.MDK_UINT_8, 2)
+			} else if pnt.Name == "cool temperature" {
+				endec.EncodeData(serialData, floatValue, endec.MDK_TEMP, 3)
+			} else if pnt.Name == "heat temperature" {
+				endec.EncodeData(serialData, floatValue, endec.MDK_TEMP, 4)
+			} else if pnt.Name == "fan speed" {
+				endec.EncodeData(serialData, uint8(floatValue), endec.MDK_UINT_8, 5)
+			} else if pnt.Name == "vertical position" {
+				endec.EncodeData(serialData, uint8(floatValue), endec.MDK_UINT_8, 6)
+			} else if pnt.Name == "horizontal position" {
+
+			}
+		}
+	}
```

Comment:
@maharjanaman can you please show tan how to get the address id and data type from the point model as per these https://github.com/NubeIO/module-core-loraraw/pull/39/files#diff-0a2c23f05a4f5c27ab49210097aac86bd8f07549b36a4a81ac069ee9cf74f449R16-R25

Reply:
@lmtan91 this is the same as the other comment re this needing to be generic/agnostic